op/reboot.go

```diff
@@ -249,48 +551,75 @@ func listProtectedNamespaces(ctx context.Context, cs *kubernetes.Clientset, ls *
 	return nss, nil
 }
 
-func checkJobPodNotExist(ctx context.Context, cs *kubernetes.Clientset, n *cke.Node) error {
+// enumeratePods enumerates Pods on a specified node.
+// It calls podHandler for each Pods not owned by Job nor DaemonSet and calls jobPodHandler for each running Pods owned by a Job.
+// If those handlers returns error, this function returns the error immediately.
+func enumeratePods(ctx context.Context, cs *kubernetes.Clientset, node string,
+	podHandler func(pod *corev1.Pod) error, jobPodHandler func(pod *corev1.Pod) error) error {
+
 	podList, err := cs.CoreV1().Pods(corev1.NamespaceAll).List(ctx, metav1.ListOptions{
-		FieldSelector: fields.SelectorFromSet(fields.Set{"spec.nodeName": n.Nodename()}).String(),
+		FieldSelector: fields.SelectorFromSet(fields.Set{"spec.nodeName": node}).String(),
 	})
 	if err != nil {
 		return err
 	}
+
 	for i := range podList.Items {
 		pod := &podList.Items[i]
 		owner := metav1.GetControllerOf(pod)
-		if owner == nil || owner.Kind != "Job" {
-			continue
+		if owner != nil {
+			switch owner.Kind {
+			case "DaemonSet":
+				continue
+			case "Job":
+				switch pod.Status.Phase {
+				case corev1.PodPending:
+				case corev1.PodSucceeded:
+				case corev1.PodFailed:
+				default:
```

Comment:
It is not clear what `default` indicates.
`Running` and `Unknown`?

Reply:
Yes. there "case"s are whitelist for silent discard.